File: app/src/main/java/com/amte/shcha/fileencryptiontest/AESEncryptDecrypt.java
package com.amte.shcha.fileencryptiontest;

import java.security.Key;

import javax.crypto.Cipher;
import javax.crypto.spec.SecretKeySpec;

/**
 * Created by kschoi on 2017-05-29.
 */
public class AESEncryptDecrypt {
    //dpfxldldkfdpvmdk
    public static String encrypt(String key, String plainInput) {
        try {
            Key secureKey = new SecretKeySpec(key.getBytes("UTF-8"), "AES");
            Cipher cipher = Cipher.getInstance("AES");

            cipher.init(Cipher.ENCRYPT_MODE, secureKey);
            byte[] encryptedData = cipher.doFinal(plainInput.getBytes("UTF-8"));
            return toHex(encryptedData);
        } catch (Exception e) {
            e.printStackTrace();
        }

        return null;
    }

    public static String decrypt(String key, String encryptedInput) {
        try {
            Key secureKey = new SecretKeySpec(key.getBytes("UTF-8"), "AES");
            Cipher cipher = Cipher.getInstance("AES");

            cipher.init(Cipher.DECRYPT_MODE, secureKey);
            byte[] plainData = cipher.doFinal(toByte(encryptedInput));
            String strignplainData = new String(plainData);
            return strignplainData;
        } catch (Exception e) {
            e.printStackTrace();
        }

        return null;
    }

    private static String toHex(byte[] buf) {
        if (buf == null) {
            return "";
        }

        StringBuffer result = new StringBuffer();
        for (byte b : buf) {
            result.append(Integer.toString((b & 0xF0) >> 4, 16));
            result.append(Integer.toString(b & 0x0F, 16));
        }
        return result.toString();
    }

    private static byte[] toByte(String hexString) {
        byte[] result = null;
        try {
            if (hexString == null) {
                return null;
            }
            int length = hexString.length();
            if (length % 2 == 1) {
                throw new IllegalArgumentException("For input string: \"" + hexString + "\"");
            }

            length = length / 2;
            result = new byte[length];
            for (int i = 0; i < length; i++) {
                int index = i * 2;
                result[i] = (byte) (Short.parseShort(hexString.substring(index, index + 2), 16));
            }

        } catch (Exception e) {
            e.printStackTrace();
        }
        return result;
    }
}

File: app/src/main/java/com/amte/shcha/fileencryptiontest/AESEncryptDecrypt.kt
package com.amte.shcha.fileencryptiontest

import java.security.Key
import javax.crypto.Cipher
import javax.crypto.spec.SecretKeySpec
import kotlin.experimental.and

/**
 * Created by kschoi on 2017-05-29.
 */
object AESEncryptDecrypt {
    //dpfxldldkfdpvmdk
    fun encrypt(key: String, plainInput: String): Unit/*String?*/ {
        try {
            val secureKey: Key = SecretKeySpec(key.toByteArray(charset("UTF-8")), "AES")
            val cipher = Cipher.getInstance("AES")
            cipher.init(Cipher.ENCRYPT_MODE, secureKey)
            val encryptedData = cipher.doFinal(plainInput.toByteArray(charset("UTF-8")))
//            return toHex(encryptedData)
        } catch (e: Exception) {
            e.printStackTrace()
        }
//        return null
    }

    fun decrypt(key: String, encryptedInput: String?): String? {
        try {
            val secureKey: Key = SecretKeySpec(key.toByteArray(charset("UTF-8")), "AES")
            val cipher = Cipher.getInstance("AES")
            cipher.init(Cipher.DECRYPT_MODE, secureKey)
            val plainData = cipher.doFinal(toByte(encryptedInput))
            return String(plainData)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }

    // error from converting java to kotlin
    /*private fun toHex(buf: ByteArray?): String {
        if (buf == null) {
            return ""
        }
        val result = StringBuffer()
        for (b in buf) {
            result.append(Integer.toString((b and 0xF0.toByte()) shr 4, 16))
            result.append(Integer.toString(b and 0x0F, 16))
        }
        return result.toString()
    }*/

    private fun toByte(hexString: String?): ByteArray? {
        var result: ByteArray? = null
        try {
            if (hexString == null) {
                return null
            }
            var length = hexString.length
            require(length % 2 != 1) { "For input string: \"$hexString\"" }
            length /= 2
            result = ByteArray(length)
            for (i in 0 until length) {
                val index = i * 2
                result[i] = hexString.substring(index, index + 2).toShort(16).toByte()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return result
    }
}